sdk/database/newdbplugin/grpc_client_test.go

```diff
@@ -0,0 +1,537 @@
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			c := gRPCClient{
+				client:  test.client,
+				doneCtx: test.doneCtx,
+			}
+
+			ctx := context.Background()
+
+			resp, err := c.NewUser(ctx, test.req)
+			test.assertErr(t, err)
+
+			if !reflect.DeepEqual(resp, test.expectedResp) {
+				t.Fatalf("Actual response: %#v\nExpected response: %#v", resp, test.expectedResp)
+			}
+		})
+	}
+}
+
+func TestGRPCClient_UpdateUser(t *testing.T) {
+	runningCtx := context.Background()
+	cancelledCtx, cancel := context.WithCancel(context.Background())
+	cancel()
+
+	type testCase struct {
+		client    proto.DatabaseClient
+		req       UpdateUserRequest
+		doneCtx   context.Context
+		assertErr errorAssertion
+	}
+
+	tests := map[string]testCase{
+		"missing username": {
+			client:    fakeClient{},
+			req:       UpdateUserRequest{},
+			doneCtx:   runningCtx,
+			assertErr: assertErrNotNil,
+		},
+		"missing changes": {
+			client: fakeClient{},
+			req: UpdateUserRequest{
+				Username: "user",
+			},
+			doneCtx:   runningCtx,
+			assertErr: assertErrNotNil,
+		},
+		"empty password": {
+			client: fakeClient{},
+			req: UpdateUserRequest{
+				Username: "user",
+				Password: &ChangePassword{
+					NewPassword: "",
+				},
+			},
+			doneCtx:   runningCtx,
+			assertErr: assertErrNotNil,
+		},
+		"zero expiration": {
+			client: fakeClient{},
+			req: UpdateUserRequest{
+				Username: "user",
+				Expiration: &ChangeExpiration{
+					NewExpiration: time.Time{},
+				},
+			},
+			doneCtx:   runningCtx,
+			assertErr: assertErrNotNil,
+		},
+		"bad expiration": {
+			client: fakeClient{},
+			req: UpdateUserRequest{
+				Username: "user",
+				Expiration: &ChangeExpiration{
+					NewExpiration: invalidExpiration,
+				},
+			},
+			doneCtx:   runningCtx,
+			assertErr: assertErrNotNil,
+		},
+		"database error": {
+			client: fakeClient{
+				updateUserErr: errors.New("update user error"),
+			},
+			req: UpdateUserRequest{
+				Username: "user",
+				Password: &ChangePassword{
+					NewPassword: "asdf",
+				},
+			},
+			doneCtx:   runningCtx,
+			assertErr: assertErrNotNil,
+		},
+		"plugin shut down": {
+			client: fakeClient{
+				updateUserErr: errors.New("update user error"),
+			},
+			req: UpdateUserRequest{
+				Username: "user",
+				Password: &ChangePassword{
+					NewPassword: "asdf",
+				},
+			},
+			doneCtx:   cancelledCtx,
+			assertErr: assertErrEquals(ErrPluginShutdown),
+		},
+		"happy path - change password": {
+			client: fakeClient{},
+			req: UpdateUserRequest{
+				Username: "user",
+				Password: &ChangePassword{
+					NewPassword: "asdf",
+				},
+			},
+			doneCtx:   runningCtx,
+			assertErr: assertErrNil,
+		},
+		"happy path - change expiration": {
+			client: fakeClient{},
+			req: UpdateUserRequest{
+				Username: "user",
+				Expiration: &ChangeExpiration{
+					NewExpiration: time.Now(),
+				},
+			},
+			doneCtx:   runningCtx,
+			assertErr: assertErrNil,
+		},
```

Comment:
I don't think it would be valuable since the statements aren't actually used anywhere. The gRPC client is faked, so the statements wouldn't go anywhere.